test/e2e/framework/framework.go

```diff
@@ -155,6 +155,16 @@ func (f *Framework) BeforeEach() {
 	if f.ClientSet == nil {
 		By("Creating a kubernetes client")
 		config, err := LoadConfig()
+		testDesc := CurrentGinkgoTestDescription()
+		switch {
+		case len(testDesc.ComponentTexts) > 0:
```

Comment:
can you just switch this to an if statement?  The switch case for a case of one makes little sense imo.